### ISSUE TITLE
Header Filters when using Mutators and Models

### ISSUE THREAD

**Issue by theref** [anvilistas/tabulator#149]
I'm trying to get Head Filters to work whilst also using the Mutator + Model combo.
Tabulator options looks like:

```python
tabulator_options = {
    "index": "uuid",
    "mutator": MyClass,
}

tabulator_columns = [
    {
        "title": "Name",
        "field": "name",
        "headerFilter": True,
    },
]
```
This correctly adds an input box but populating that box has no effect on the table.

**Comment by s-cork** [comment]
are you also using an app_table?

**Comment by theref** [comment]
Yes, thanks for the pointer, I'd forgotten about that: `self.tabulator.options["app_table"] = anvil.server.call("get_everything")` the result of that server function is : `return app_tables.my_class.client_readable()`

**Comment by s-cork** [comment]
you'll need to completely customize filtering 
You can't use the builtin header filters because these don't know how to filter lazy data
it's possibly a feature we could add
but - i didn't add it because it wasn't obvious how to make it work in practice


from the docs

```
import anvil.tables.query as q

def text_box_1_change(self, **event_args):
    name = self.text_box_1.text
    if not name:
        self.tabulator.clear_query()
    else:
        self.tabulator.set_query(name=q.ilike(f"%{name}%"))
```

So you can adapt that as you like

the issue you'll have is adding the convenience of a textbox to the header
that's a bit of a faff - so if you can move the textbox away from the tabulator element, then you're good

**Comment by theref** [comment]
this worked perfectly, thank you
